Makefile

```diff
@@ -0,0 +1,48 @@
+REGISTRY=staging-k8s.gcr.io
+IMAGE=trace-context-injector
+
+clean:
+	@echo 'Removing generated configurations and local image build...'
+	rm deploy/mutatingwebhook-ca-bundle.yaml
+	docker rmi -f $(IMAGE)
+
+# docker commands
+docker-build:
+	@echo 'Building image $(IMAGE)...'
+	docker build -t $(IMAGE) .
+
+docker-release: docker-build docker-publish
+
+docker-publish: docker-tag-latest
+	@echo 'Pushing image $(IMAGE) to $(REGISTRY)...'
+	docker push $(REGISTRY)/$(IMAGE):latest
+
+docker-tag-latest:
```

Comment:
tagging and pushing to latest is probably a bit fine-grained.  I would prefer just having `make docker` build a docker image, and document the release process in a .md document.  The release documentation can be added later

Reply:
to expand on this, we should only add things to the makefile if we only expect a single variant.  I.E. we should avoid parameterized makefiles wherever possible

Reply:
done, will document release process in later PR